src/moonlink/src/storage/iceberg/tests.rs

```diff
@@ -1906,7 +1919,7 @@ async fn mooncake_table_snapshot_persist_impl(iceberg_table_config: IcebergTable
     // Check iceberg snapshot store and load, here we explicitly load snapshot from iceberg table, whose construction is lazy and asynchronous by design.
     let mut iceberg_table_manager = IcebergTableManager::new(
         mooncake_table_metadata.clone(),
-        ObjectStorageCache::default_for_test(&cache_temp_dir), // Use separate and fresh new cache for each recovery.
+        cache.clone(), // Use separate and fresh new cache for each recovery.
```

Comment:
This requires some context here, so for every iceberg table manager you should create new object storage cache, otherwise it has conflict with other usage.

What I would suggest is to create a test util function at https://github.com/Mooncake-Labs/moonlink/blob/main/src/moonlink/src/storage/mooncake_table/table_creation_test_utils.rs
which does similar thing as `ObjectStorageCache::default_for_test` but returns `Arc<dyn CacheTrait>`.